pyramid_openapi3/exceptions.py

```diff
@@ -0,0 +1,39 @@
+"""Exceptions used in pyramid_openapi3."""
+from pyramid.httpexceptions import HTTPBadRequest
+from pyramid.httpexceptions import HTTPInternalServerError
+
+import typing as t
+
+
+class RequestValidationError(HTTPBadRequest):
+
+    explanation = "Request validation failed."
+
+    def __init__(self, *args, errors, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.errors = errors
+        self.detail = self.message = "\n".join(str(e) for e in errors)
```

Comment:
should be deferred to handler

Reply:
You mean the exception view handler? The pyramid default exception view handler does not handle exceptions with an `error` attribute. Are you suggesting to add a default exception handler to this library?

Reply:
No, the exact line ```self.detail = self.message = "\n".join(str(e) for e in errors)``` you are creating object in memory that are not even needed yet.

Reply:
I was trying to keep the pyramid httpexception contract up here with some useful information from validation errors, and not re-implementing all the templating that's implemented in pyramid http exception/response. 

I'll have a think about whether it's possible to delay that `detail` / `message` generation. The only way I see for now is to use a `@property` wrapper around `.message` and `.detail` . Maybe there are better ways?